src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/Runners/AppRunner.cs

```diff
@@ -167,6 +169,16 @@ public async Task StartAsync(CancellationToken token, bool waitForReady = true)
                     ProfilerIdentifiers.EnvironmentVariables.StdErrLogger_Level, ProfilerLogLevel);
             }
 
+            if (EnableMonitorStartupHook)
+            {
+                string startupHookPath = AssemblyHelper.GetAssemblyArtifactBinPath(
```

Comment:
Move this to the ctor so that you can calculate the path relative to the test assembly rather than relative to this utility assembly; it's likely trying to take something like `artifacts/bin/<TestAssembly>/Debug/net8.0/<TestCommonAssembly>.dll` and turning it into `artifacts/bin/<TestAssembly>/Debug/net8.0/<StartupHookAssembly>.dll`, which is not correct.

Reply:
So, I can change this if it's broken, but this was actually a direct copy from the `feature/parameter-capturing` branch https://github.com/dotnet/dotnet-monitor/pull/5072#discussion_r1283743205